### PR TITLE
Search result

### PR DESCRIPTION
**Pull request by BBriset**
Set search bar functional and build resultCard component who render the results of the querys. First commits have not been saved,  i thought they was :( , i realised that after the push...). ResulCard handle the case when there is no poster returned by the api

### CODE REVIEW

**Review by BenZen59 — APPROVED** [review]

**Review by BenZen59 — APPROVED** [review]

**Review by BenZen59 — APPROVED** [review]

**Review by BenZen59 — DISMISSED** [review]

File: src/components/searchBar/SearchBar.jsx [hunk 1]
```diff
@@ -1,16 +1,49 @@
-import React from 'react';
+import Axios from 'axios';
+import React, { useState } from 'react';
+import ResultCard from '../ResultCard/ResultCard';
 import './Search.css';
 
-function Search() {
+const Search = () => {
+  const [querys, setQuerys] = useState('');
```
Comment (LoicBrassart) [inline src/components/searchBar/SearchBar.jsx]: Querys ? Plural ? Where's my array ?

File: src/components/searchBar/SearchBar.jsx [hunk 2]
```diff
@@ -1,16 +1,49 @@
...
+  const onChange = (e) => {
+    e.preventDefault();
+    setQuerys(e.target.value);
+    Axios.get(
+      `https://api.themoviedb.org/3/search/movie?api_key=599ded6f0fc3bcaee1882e83ae0d438a&language=en-US&page=1&include_adult=false&query=${e.target.value}`
+    )
+      .then((response) => response.data)
+      .then((data) => setResults(data.results))
+      .catch(() => {
+        console.error('Erreur API');
```
Comment (LoicBrassart) [inline src/components/searchBar/SearchBar.jsx]: No `console` on `dev` !

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Minor Evils to fix

**Comment by BBriset** [comment]
fixed :)

File: src/components/ResultCard/ResultCard.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import './ResultCard.css';
+
+const ResultCard = (movie) => {
+  const { title, poster_path: posterPath } = movie;
+  const linkToFicheFilm = (e) => {
```
Comment (Hacene010) [inline src/components/ResultCard/ResultCard.jsx]: No varaiable's names in french please !!

File: src/components/ResultCard/ResultCard.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    e.preventDefault();
+  };
+  return (
+    <div
+      className="resultcard"
```
Comment (Hacene010) [inline src/components/ResultCard/ResultCard.jsx]: respect the camelCase everywhere plz

File: src/components/searchBar/SearchBar.jsx [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
   return (
-    <div className="Search">
-      <input
-        className="SearchBar"
-        type="search"
-        placeholder="Search your movie"
-      />
-    </div>
+    <>
+      <div className="Search">
```
Comment (Hacene010) [inline src/components/searchBar/SearchBar.jsx]: camelCase !

File: src/components/searchBar/SearchBar.jsx [hunk 6]
```diff
@@ -1,16 +1,48 @@
...
+        <input
+          className="SearchBar"
```
Comment (Hacene010) [inline src/components/searchBar/SearchBar.jsx]: camelCase too :)

**Review by Hacene010 — CHANGES_REQUESTED** [review]
still some syntax fix see the comments

**Comment by BBriset** [comment]
Hope we good now,  i haven't seen anything anything more.

**Review by Hacene010 — DISMISSED** [review]
That's better ;)

**Comment by LoicBrassart** [comment]
No validation from SM

**Review by BenZen59 — DISMISSED** [review]

File: src/components/ResultCard/ResultCard.jsx [hunk 7]
```diff
@@ -3,15 +3,15 @@ import './ResultCard.css';
...
   };
   return (
     <div
-      className="resultcard"
+      className="resultCard"
       role="button"
-      onClick={linkToFicheFilm}
-      onKeyPress={linkToFicheFilm}
+      onClick={linkToMovieProfile}
+      onKeyPress={linkToMovieProfile}
```
Comment (LoicBrassart) [inline src/components/ResultCard/ResultCard.jsx]: So, this component (`div`)... 
- Must do nothing when we click on it ? 
- And you wrote 2 EListeners to ensure this ?
- And it's calling a "LinkBlabla" function, but there is absolutely no `<a>` nor `<Link>` anywhere in sight ?
- As long as I'm here : why don't you use a `button` ?

File: src/components/searchBar/SearchBar.jsx [hunk 8]
```diff
@@ -4,31 +4,30 @@ import ResultCard from '../ResultCard/ResultCard';
 import './Search.css';
 
 const Search = () => {
-  const [querys, setQuerys] = useState('');
+  const [query, setQuery] = useState(['']);
```
Comment (LoicBrassart) [inline src/components/searchBar/SearchBar.jsx]: Nope!
If `query` is singular (which seems legit) it should *never* contain an Array

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some more Evils to fix

File: src/components/searchBar/SearchBar.jsx (hunk 8 above)
Comment (BBriset) [reply src/components/searchBar/SearchBar.jsx]: Oops i havent understood your comment like this

File: src/components/ResultCard/ResultCard.jsx (hunk 7 above)
Comment (BBriset) [reply src/components/ResultCard/ResultCard.jsx]: For now its dont have any purpose than set the comportement, its set out for link to another components/page when we will set up the route soon. And for the Key listener thanks eslint Who have contrain me to ad it for validation. No button cause We dont wan't an ugly one on our cardResult especialy on mobile version.

**Review by BenZen59 — DISMISSED** [review]

**Review by Hacene010 — DISMISSED** [review]
Seems ok for me

**Review by Quentin-Macquart — APPROVED** [review]
It seems ok to me!

**Review by Hacene010 — APPROVED** [review]
ok for me

**Review by LoicBrassart — APPROVED** [review]

**Pull request merged**